rust/src/image_processing.rs

```diff
@@ -234,12 +234,13 @@ pub(super) fn remove_boundary_artifact(
 
     // We want one dimension of the output to be 256 and we we want the aspect ratio of the output
     // to match the input image.
+    let other_dim;
     let mut mean_padded: ndarray::Array4<f32> = if width > height {
-        let other = ((width as f32 / height as f32) * 256.0) as usize;
-        ndarray::Array4::zeros([1, 3, 256_usize, other])
+        other_dim = ((width as f32 / height as f32) * 256.0) as usize;
+        ndarray::Array4::zeros([1, 3, 256_usize, other_dim])
     } else {
-        let other = (height / width) * 256;
-        ndarray::Array4::zeros([1, 3, other, 256])
+        other_dim = ((height as f32 / width as f32) * 256.0) as usize;
```

Comment:
Not sure if this is relevant here, but do you always want to round down after this operation?

Reply:
Yep rounding towards zero is intentional.